src/arrayMethodReduce.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result;
+
+    if (arguments.length < 2) {
+      result = this[0];
+
+      for (let i = 1; i < this.length; i++) {
+        result = callback(result, this[i], i, this);
+      }
+    } else {
+      result = initialValue;
+
+      for (let i = 0; i < this.length; i++) {
+        result = callback(result, this[i], i, this);
+      }
+    }
+
+    return result;
```

Comment:
create an variable startIndex 0 by default and change its value if second argument is not passed to avoid duplication